consensus/types/src/beacon_state.rs

```diff
@@ -2214,12 +2169,10 @@ impl<E: EthSpec> BeaconState<E> {
             .validators_mut()
             .get_mut(validator_index)
             .ok_or(Error::UnknownValidator(validator_index))?;
-        if validator.has_eth1_withdrawal_credential(spec) {
-            AsMut::<[u8; 32]>::as_mut(&mut validator.withdrawal_credentials)[0] =
-                spec.compounding_withdrawal_prefix_byte;
+        AsMut::<[u8; 32]>::as_mut(&mut validator.withdrawal_credentials)[0] =
+            spec.compounding_withdrawal_prefix_byte;
```

Comment:
hmmm we should probably refactor the [`change_withdrawal_credentials()`](https://github.com/sigp/lighthouse/blob/v5.3.0/consensus/types/src/validator.rs#L158) method on the `Validator` so that it accepts a prefix.

Reply:
This way seems cleaner to me tbh, we are changing it in place instead of creating the entire byte array from scratch

Reply:
Can the `change_withdrawal_credentials()` function not be refactored to also do it in place?

Reply:
that function also changes the execution address, not just the prefix. Can do it, but i'm not sure what it adds 🤷‍♂️

Reply:
Gotcha. I guess I'm just arguing for encapsulating it as a function on the validator. Honestly just realizing the naming of this sucks. `0x01` is `eth1` credentials (despite wanting to move away from "eth1" and towards "execution") but "execution" now means and both `0x01` and `0x02` 😩

I suppose we could fix this in a future PR.. but like what do you think about being explicit:
`eth1` -> `0x01` for example `has_0x01_withdrawal_credentials`

and we could rename `change_withdrawal_credentials` to `change_to_0x01_credentials` and add a new function `change_to_0x02_credentials` which encapsulates what you have above. `has_execution_withdrawal_credentials` remains the same and is more consistent now.

Reply:
Yeah I agree the naming is super confusing, agree with doing the renaming in a future PR